src/domUpdates.js

```diff
@@ -10,28 +10,45 @@ let tripsHeader = $('#trips-header');
 let spendHeader = $('#spend-header');
 let totalSpend = $('#total-spend');
 
+tripsStatus.on('click', (event) => domUpdates.updateTripStatus(event));
+
 let domUpdates = {
+
   displayTravelerInfo() {
     Promise.all([this.getTravelerData(), this.getTripsData(), this.getDestinationsData()])
       .then(data => {
         this.greetUser(data[0])
-        const traveler = this.instantiateTraveler(data)
+        const travelerData = data[0];
+        const tripsData = data[1];
+        const destinationsData = data[2];
+        const traveler = this.instantiateTraveler(travelerData, tripsData, destinationsData)
         this.displayTrips(traveler.trips);
         this.displayCost(traveler);
       })
       // .catch(error => console.log(error.message));
   },
 
-  instantiateTraveler(data) {
-    const travelerData = data[0];
-    const tripsData = data[1];
-    const destinationsData = data[2];
+  displayAgentInfo() {
+    Promise.all([this.getAllTravelersData(), this.getTripsData(), this.getDestinationsData()])
+      .then(data => {
+        this.greetAgent();
+        const allTravelersData = data[0];
+        const tripsData = data[1];
+        const destinationsData = data[2];
+        const allTravelers = allTravelersData.map((travelerData) => {
+          return this.instantiateTraveler(travelerData, tripsData, destinationsData)
+        })
```

Comment:
I'm not 100% sure about this, but I think there might be a way to consolidate these two promise blocks and the fetch below into one fetch/promise that happens on load. It could simplify the code a bit!

Reply:
Plus, if you consolidate your promises and fetches into one function, it will be easier to test and ID when something doesn't work properly.